### PR TITLE
One list of GalleryItems to rule them all

### PR DESCRIPTION
**Pull request by HansMuller**
GalleryItem and kAllGalleryItems completely now describe the gallery's home page well enough for the smoke and performance test.

### CODE REVIEW

File: examples/flutter_gallery/lib/gallery/item.dart [hunk 1]
```diff
@@ -34,3 +39,176 @@ class GalleryItem extends StatelessWidget {
...
+  new GalleryItem(
+    title: 'List',
+    routeName: ListDemo.routeName,
+    buildRoute: (BuildContext) => new ListDemo()
+  ),
+  new GalleryItem(
+    title: 'Menus',
+    routeName: MenuDemo.routeName,
+    buildRoute: (BuildContext) => new MenuDemo()
+
```
Comment (abarth) [inline examples/flutter_gallery/lib/gallery/item.dart]: Spurious blank line.

**Comment by abarth** [comment]
LGTM

File: examples/flutter_gallery/test/smoke_test.dart [hunk 2]
```diff
@@ -2,23 +2,22 @@
...
 import 'package:flutter_gallery/main.dart' as flutter_gallery_main;
 
-// Warning: the following strings must be kept in sync with GalleryHome.
-const List<String> demoCategories = const <String>[
-  'Demos',
-  'Components',
-  'Style'
-];
+final List<String> demoCategories = new LinkedHashSet.from(
+  kAllGalleryItems.map((GalleryItem item) => item.category)).toList();
```
Comment (Hixie) [inline examples/flutter_gallery/test/smoke_test.dart]: newline before the last )

**Comment by Hixie** [comment]
LGTM

**Comment by yjbanov** [comment]
I think this PR broke the integration test: https://github.com/flutter/flutter/issues/4555

**Pull request merged**